advanced/security.md

```diff
@@ -2,46 +2,141 @@
->  ```html
->  media-src *;
->  ```
+>
+> ```html
+> media-src 'self' *.youtube.com;
+> ```
+>
+> To allow content from any source, use the (\*) wildcard. Allowing all sources (using \*) negates the security policy for the source type. For example:
+>
+> ```html
+> media-src *;
+> ```
+>
 > For information on Content Security Policies, see: [MDN Web Docs - Content Security Policy (CSP)](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP).
 
 When using the {{site.cloudname}}, use this CSP header :
 
 ```html
-<meta http-equiv="Content-Security-Policy" content="default-src 'none'; script-src 'self' *.tinymce.com *.tiny.cloud; connect-src 'self' *.tinymce.com *.tiny.cloud blob:; img-src 'self' *.tinymce.com *.tiny.cloud data: blob:; style-src 'self' 'unsafe-inline' *.tinymce.com *.tiny.cloud; font-src 'self' *.tinymce.com *.tiny.cloud;" />
+<meta
+  http-equiv="Content-Security-Policy"
+  content="default-src 'none'; script-src 'self' *.tinymce.com *.tiny.cloud; connect-src 'self' *.tinymce.com *.tiny.cloud blob:; img-src 'self' *.tinymce.com *.tiny.cloud data: blob:; style-src 'self' 'unsafe-inline' *.tinymce.com *.tiny.cloud; font-src 'self' *.tinymce.com *.tiny.cloud;"
+/>
 ```
 
-When self-hosting {{site.productname}} from a local domain, use this CSP header (excludes the &#42;.tiny.cloud domain):
+When self-hosting {{site.productname}} from a local domain, use this CSP header (excludes the \*.tiny.cloud domain):
 
 ```html
-<meta http-equiv="Content-Security-Policy" content="default-src 'none'; script-src 'self'; connect-src 'self' blob:; img-src 'self' data: blob:; style-src 'self' 'unsafe-inline'; font-src 'self';" />
+<meta
+  http-equiv="Content-Security-Policy"
+  content="default-src 'none'; script-src 'self'; connect-src 'self' blob:; img-src 'self' data: blob:; style-src 'self' 'unsafe-inline'; font-src 'self';"
+/>
 ```
+
+## Potential security risks and mitigation measures
+
+The following security risks are not {{site.productname}} specific, but are the main security risks associated with websites or applications which allow user inputs. Protecting your services and users from these risks requires server-side handling. Note that attackers will likely bypass any editor and attack the server directly, rather than use the text editor as a vector.
+
+For information on mitigating these risks, see the provided links in each section.
+
+### Cross-Site Scripting (XSS)
+
+Cross-Site Scripting (XSS) involves using website or application inputs to inject malicious, client-side code. This code can then be used to attack your users.
+
+Although {{site.productname}} removes most XSS content, precautions should be taken to protect your users, such as enabling a Content Security Policy (CSP) as described in [Configuring Content Security Policy (CSP) for TinyMCE](#configuringcontentsecuritypolicycspfortinymce).
```

Comment:
I'd also add the HTML purify alternative you mentioned earlier, as not all users will be able to use a CSP. Even then it should be used in combination with a server-side santization library, as they could bypass the CSP and post directly to the server or since we require `unsafe-inline` for styles, an attacker could still trigger an XSS attack if TinyMCE was vulnerable.

So to sum up, we should be recommending a combination of things for preventing XSS, rather than the single CSP option (so basically the XSS/Sanitzation and Further information headings is all I'd keep).

Reply:
changed to:
> precautions should be taken to protect your users, such as enabling a Content Security Policy (CSP) and server-side filtering.